lib/active_model/serializer/adapter/json/fragment_cache.rb

```diff
@@ -1,16 +1,6 @@
-require 'active_model/serializer/adapter/fragment_cache'
-module ActiveModel
-  class Serializer
-    class Adapter
-      class Json < Adapter
-        class FragmentCache
+class ActiveModel::Serializer::Adapter::Json::FragmentCache
```

Comment:
I've tried to remove the inherited hook, and replace it with a method in the Adapter class itself, but even then, I'd need to prefix **every** file that may define `class Adapter` rather than reopen it, with something like `adapter = adapter = ActiveModel::Serializer::Adapter # Need to load the Adapter before opening it below`

Not even `Adapter.eager_load!` is sufficient.